tests/testthat/test_point_estimation.R

```diff
@@ -65,19 +64,119 @@ test_that("Does monte_carlo function give benchmark results?", {
                                       gen_model = gen_par
                                       )
 
+  set.seed(100)
+  point2 <- point_estim(framework = framework,
+                        fixed = income ~ educ1,
+                        transformation = "box.cox",
+                        interval = "default",
+                        L = 2,
+                        keep_data = FALSE)
+
   # Load benchmark point estimates
-  ebp_point_bc <- read.csv2("EBP/ebp_point_bc.csv", sep = ",", 
+  ebp_point_bc <- read.csv2("EBP/ebp_point_bc.csv", sep = ",",
                             stringsAsFactors = TRUE)
- 
+
   # compare 10% quantile
   expect_equal(point[,"Quantile_10"],
                as.numeric(as.character(ebp_point_bc[,"quant10"])))
+  expect_equal(point2$ind[,"Quantile_10"],
+               as.numeric(as.character(ebp_point_bc[,"quant10"])))
+
   # compare HCR
   expect_equal(point[,"Head_Count"],
                as.numeric(as.character(ebp_point_bc[,"hcr"])))
-
+  expect_equal(point2$ind[,"Head_Count"],
+               as.numeric(as.character(ebp_point_bc[,"hcr"])))
 
 })
 
 
+test_that("Does monte_carlo function give benchmark results? Using weights
+          and pop_weights", {
+  suppressWarnings(RNGversion("3.5.0"))
+  # join synthetic population weights
+  Xoutsamp_AuxVar$pop_weights <- rep(2, nrow(Xoutsamp_AuxVar))
+
+  # Single elements needed in monte_carlo()
+  framework <- framework_ebp(fixed = income ~ educ1,
+                             pop_data = Xoutsamp_AuxVar, pop_domains = "provlab",
+                             smp_data = incomedata, smp_domains = "provlab",
+                             threshold = 4282.081,
+                             custom_indicator = NULL,
+                             na.rm = TRUE,
+                             aggregate_to = "domain",
+                             weights = "weight",
+                             pop_weights = "pop_weights")
+
+  # Fixed optimal parameter and shift (benchmark values)
```

Comment:
If this is the test with weights and population weights, why the Box-Cox transformation, that is not possible, is it? Is there a test for aggregate_to?

Reply:
Test with different aggregation level added. Corrections concerning transformation.